homeassistant/components/flexit/climate.py

```diff
@@ -64,7 +64,7 @@ def __init__(self, hub, modbus_slave, name):
         self._heating = None
         self._cooling = None
         self._alarm = False
-        self.unit = pyflexit(hub, modbus_slave)
+        self.unit = pyflexit(hub._client, modbus_slave)
```

Comment:
Referencing the _client in the modbushub have a number of side-effects which I suppose is unwanted:
- _client is not guaranteed to point at a serial client, it can very well be None.
- you bypass all the error handling in the modbus integration, and would be better of dealing directly with pymodbus.

The real problem is that modbus is now async and thus the calls have changed. But since the call (I suppose one among a number) is hidden in the library pyflexit, I do not have suggestions on how to bypass that.

The owner of pyflexit needs to update the library, or even better inherit modbus climate in the flexit integration and add the flexit specialities.

Sorry I cannot be of much help here.